elastic/eui
# [Visual Refresh] Update vis colors and palettes

## Summary

closes https://github.com/elastic/eui-private/issues/127

>[!IMPORTANT]
This PR merges into a feature branch.

>[!NOTE]
The first commit of this PR is duplicated from this other [PR](https://github.com/elastic/eui/pull/8111) to ensure shared setup. This commit should not be review on this PR but the previous one.

This PR adds theme specific visualization colors and updates eui color palettes to use theme-specific colors.

## Changes

- adds global `EuiVisColorStore` - stores current theme vis colors on `EuiProvider` theme change 
  - EUI color palettes previously were using static values and are just - non-react - Javascript; they might be used outside of a react context
  - to support current and new theme this global store provides means to add reactivity to the previously static usage
  - provides pub/sub pattern to ensure consumers can update values on theme update, even outside of the react provider context
  
![Screenshot 2024-11-04 at 10 57 22](https://github.com/user-attachments/assets/1a0afa86-29d5-4e5d-8f56-6d54f39eb7de)
  
- adds new theme vis color tokens

![Screenshot 2024-11-06 at 23 02 42](https://github.com/user-attachments/assets/d626dc68-decf-4345-b2de-3730788b34f8)

- updates EUI color palettes to use vis color tokens stored in the `EuiVisColorStore` instead of static values
- updates usages of EUI color palettes
- migrated "Color palettes" docs to EUI+

 #unrelated
- changes EUI docs theme switcher, updates are taken from [this previous PR](https://github.com/elastic/eui/pull/8036)

## QA

- verify color palette usages update colors on theme change:
 - [ ] EUI docs: https://eui.elastic.co/pr_8112/#/forms/color-selection
 - [ ] EUI+ docs: https://eui.elastic.co/pr_8112/new-docs/docs/color_palettes/
   - https://eui.elastic.co/pr_8112/storybook/index.html?path=/story/forms-euicolorpalettepicker-euicolorpalettedisplay--playground
   - https://eui.elastic.co/pr_8112/storybook/index.html?path=/story/forms-euicolorpalettepicker-euicolorpalettepicker--playground&globals=theme:EUI_THEME_BOREALIS
   - https://eui.elastic.co/pr_8112/storybook/index.html?path=/story/forms-euicolorpicker-euicolorpicker--playground
